### PR TITLE
[PPM-286] Remove AGENT_PING_REQUEST/RESPONSE

### PR DESCRIPTION
**Pull request by VladyslavTupikin**
Need to remove next messages and all related source code:

```
ACTION_CONTROL_AGENT_PING_REQUEST
ACTION_CONTROL_AGENT_PING_RESPONSE
```

The main reason is: these vendor-specific messages no longer be useful and should be removed.

https://jira.prplfoundation.org/browse/PPM-286

### CODE REVIEW

**Comment by morantr** [comment]
Please do all the PING message removal commits on a single PR (#1534).

**Comment by VladyslavTupikin** [comment]
@morantr  I have 2 separate issues, created 2 separate pull requests.

File: agent/src/beerocks/slave/son_slave_thread.cpp [hunk 1]
```diff
@@ -251,53 +251,6 @@ bool slave_thread::work()
...
-                return;
-            }
-
-            request->total() = 1;
-            request->seq()   = 0;
-            request->size()  = 0;
-
-            send_cmdu_to_controller(cmdu_tx);
-            keep_alive_retries++;
-            master_last_seen = now;
```
Comment (mariomaz) [inline agent/src/beerocks/slave/son_slave_thread.cpp:311]: I think `keep_alive_retries`and `master_last_seen` member variables are not used any more and should be removed too

File: agent/src/beerocks/slave/son_slave_thread.cpp [hunk 2]
```diff
@@ -251,53 +251,6 @@ bool slave_thread::work()
     return true;
 }
 
-void slave_thread::process_keep_alive()
-{
-    if (!config.enable_keep_alive || !son_config.slave_keep_alive_retries ||
```
Comment (mariomaz) [inline agent/src/beerocks/slave/son_slave_thread.cpp:271]: I think `enable_keep_alive` and `slave_keep_alive_retries` configuration attributes are not used any more and should be removed too

**Review by mariomaz — APPROVED** [review]
I think some variables and configuration parameters are not used any more after removing the PING messages and thus they should be removed too.

File: agent/src/beerocks/slave/son_slave_thread.cpp (hunk 2 above)
Comment (VladyslavTupikin) [reply agent/src/beerocks/slave/son_slave_thread.cpp:271]: fixed

File: agent/src/beerocks/slave/son_slave_thread.cpp (hunk 1 above)
Comment (VladyslavTupikin) [reply agent/src/beerocks/slave/son_slave_thread.cpp:311]: fixed

File: framework/platform/bpl/include/bpl/bpl_err.h [hunk 3]
```diff
@@ -45,7 +45,6 @@ namespace bpl {
 /* 14 */ ERROR_CODE(BH_SLAVE_SOCKET_DISCONNECTED)                                           \
 /* 15 */ ERROR_CODE(BH_STOPPED)                                                             \
 /* 16 */ ERROR_CODE(SLAVE_CONNECTING_TO_BACKHAUL_MANAGER)                                   \
-/* 17 */ ERROR_CODE(SLAVE_MASTER_KEEP_ALIVE_TIMEOUT)                                        \
 /* 18 */ ERROR_CODE(SLAVE_INVALID_MASTER_SOCKET)                                            \
```
Comment (arnout) [inline framework/platform/bpl/include/bpl/bpl_err.h]: Note that this changes the numbering of everything that comes after....

File: common/beerocks/bcl/include/bcl/beerocks_config_file.h [hunk 4]
```diff
@@ -74,7 +74,6 @@ class config_file {
         std::string monitor_ap_active_threshold_B;
         std::string monitor_ap_idle_stable_time_sec;
         std::string monitor_disable_initiative_arp;
-        std::string slave_keep_alive_retries;
```
Comment (arnout) [inline common/beerocks/bcl/include/bcl/beerocks_config_file.h:77]: (nitpick) the commit that remove this variable could be squashed together, with a commit log like this:

```
agent: remove deprecated slave_keep_alive_retries

Also remove the corresponding variables from bcl and controller.
```

**Review by arnout — APPROVED** [review]

File: framework/platform/bpl/include/bpl/bpl_err.h (hunk 3 above)
Comment (VladyslavTupikin) [reply framework/platform/bpl/include/bpl/bpl_err.h]: Fixed commented numeration for rest of error codes.

File: common/beerocks/bcl/include/bcl/beerocks_config_file.h (hunk 4 above)
Comment (VladyslavTupikin) [reply common/beerocks/bcl/include/bcl/beerocks_config_file.h:77]: Fixed

**Pull request merged**